test/e2e/runner/infra.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"context"
+
+	e2e "github.com/tendermint/tendermint/test/e2e/pkg"
+)
+
+// TestnetInfraAPI provides an API for interacting with the infrastructure for
+// an entire testnet.
+type TestnetInfraAPI interface {
+	// GenerateConfig generates any necessary configuration for the
+	// infrastructure provider during testnet setup.
+	GenerateConfig() error
+
+	// Stop attempts to stop the entire testnet.
+	Stop(ctx context.Context) error
+
+	// Pause attempts to pause the entire testnet.
+	Pause(ctx context.Context) error
+
+	// Unpause attempts to resume a paused testnet.
+	Unpause(ctx context.Context) error
+
+	// ShowLogs will print all logs for the whole testnet to stdout.
+	ShowLogs() error
+
+	// TailLogs tails the logs for all nodes in the testnet.
+	TailLogs() error
+
+	// Cleanup stops and destroys all running infrastructure and deletes any
+	// generated files.
+	Cleanup() error
+}
+
+// NodeInfraAPI provides an API for interacting with specific nodes'
+// infrastructure.
+type NodeInfraAPI interface {
+	// ProvisionNode attempts to provision infrastructure for the given node
+	// and starts it.
+	ProvisionNode(ctx context.Context, node *e2e.Node) error
+
+	// DisconnectNode attempts to ensure that the given node is disconnected
+	// from the network.
+	DisconnectNode(ctx context.Context, node *e2e.Node) error
+
+	// ConnectNode attempts to ensure that the given node is connected to the
+	// network.
+	ConnectNode(ctx context.Context, node *e2e.Node) error
+
+	// KillNode attempts to ensure that the given node's process is killed
+	// immediately using SIGKILL.
+	KillNode(ctx context.Context, node *e2e.Node) error
```

Comment:
```suggestion
	Kill(ctx context.Context, node *e2e.Node) error
```
We can drop the word `Node` from these. A node is the only parameter of the method and also included in the name of the type.

Reply:
See https://github.com/tendermint/tendermint/pull/8754#discussion_r897198789